src/dotnet/ToolManifest/ToolManifestFile.cs

```diff
@@ -113,6 +115,52 @@ public bool TryFind(ToolCommandName toolCommandName, out ToolManifestPackage too
             return false;
         }
 
+        public void Add(
```

Comment:
have you considered keeping the finder, whose only job is to find the manifest and return the manifest object and then have a manifest object responsible to add/remove packages?

The way this is going, it feels like this class will end up doing too much.

Reply:
i could separate the API of _find_ vs _edit_, however _find_ will depend on _edit_ since _find_ will need to read the context for isRoot etc.

there is no "manifest object" today. the return of _FindAllManifest_ is an **combined** array of manifest **entry** (individual packages). Since combine logic could be decided by the content of manifest alone

Reply:
separate finder and editor